src/server/routers/documents.ts

```
@@ -170,51 +209,82 @@ export const documentsRouter = createTRPCRouter({
+              const chunkInserts = chunks.map((chunk: DocumentChunk) => ({
+                document_id: document.id,
+                content: chunk.content,
+                chunk_index: chunk.chunkIndex,
+                metadata: chunk.metadata,
+              }));
+
+              const { error: chunksError } = await ctx.supabaseAdmin
+                .from('document_chunks')
+                .insert(chunkInserts);
+
+              if (chunksError) {
+                throw new Error(`Failed to store chunks: ${chunksError.message}`);
+              }
+
+              // Update document with chunk count and mark as completed
               await ctx.supabaseAdmin
                 .from('documents')
                 .update({
-                  chunk_count: chunkCount,
+                  chunk_count: chunks.length,
+                  processing_status: 'completed',
                   updated_at: new Date().toISOString(),
                 })
                 .eq('id', document.id);
 
-              totalChunks += chunkCount;
-              totalTokens += Math.ceil(doc.content.length / 4);
+              totalChunks += chunks.length;
+              const docTokens = chunks.reduce((sum: number, chunk: DocumentChunk) => sum + Math.ceil(chunk.content.length / 4), 0);
+              totalTokens += docTokens;
 
               results.push({
                 title: doc.title,
                 success: true,
                 documentId: document.id,
-                chunkCount,
+                chunkCount: chunks.length,
+                tokens: docTokens,
               });
-            } catch (ragError) {
-              console.error(`RAG processing error for ${doc.title}:`, ragError);
+
+              console.log(`Successfully processed bulk document: ${doc.title} with ${chunks.length} chunks`);
+
+            } catch (chunkingError) {
+              console.error(`Chunking failed for document: ${doc.title}`, chunkingError);
```

We should be throwing error constructs rather than just console logging them. This is a silent failure that would almost be impossible to debug if deployed

---

Bet will change this